Fig S1: Cell and gene metrics comparison between Alevin-fry and Cell Ranger

allyhawkins opened the pull request:
Closes #4 

This PR adds the remaining plots needed for the benchmarking supplemental figure. 

For the plot itself: 

- I created 3 separate plots for now. I think we may want to combine them, but I want to get some feedback on the actual plot layouts before I do that. These 3 plots mirror the plots found in [FAQ: Why did we use Alevin-fry for processing?](https://scpca.readthedocs.io/en/stable/faq.html#why-did-we-use-alevin-fry-for-processing) and include a comparing the total UMI/ cell, genes detected/cell and mean gene expression across methods. 
- I did update the theme here to be consistent with the other faceted plots in the paper. I don't think we need the grid, which is the largest change. 
- For the UMI/cell and genes detected/cell I tried something where I separated out the run ID and the seq unit so that each row would represent a different seq unit and each column a library. To do this, I did have to split the dataframe I was using for plotting by seq unit and then make a separate faceted plot for each and stack them. `facet_grid` doesn't like having to make a second row with different facet labels if you are using two groups to facet by. I searched a lot for solutions to this, but this seemed to be the best approach. Do we like this layout? And if so, I'm definitely open to more efficient ways to do it! 
<img width="1411" alt="Screenshot 2024-01-17 at 12 50 28 PM" src="https://github.com/AlexsLemonade/scpca-paper-figures/assets/54039191/d8286f1a-1013-412c-bf79-8bc3128b9c05">

- For comparing the mean gene expression per cell, I tried a slightly different approach where I used `facet_wrap` but with two different labels. This shows both the run ID and the seq unit in the label. I just felt like it wasn't as obvious which samples are cell vs. nuclei. What do we think of this vs. the one I used above? One thought it to use this approach for all the plots, but color the labels distinctly for cell vs. nuclei? 
<img width="1438" alt="Screenshot 2024-01-17 at 12 52 12 PM" src="https://github.com/AlexsLemonade/scpca-paper-figures/assets/54039191/f70e4de1-3132-4a98-946a-75d7112ea63d">

I'm curious which version of these we like better? I think we want to be consistent with how we label run ID and seq unit across all three plots. Or do we want to go a completely different direction than what's here. Tagging @jaclyn-taroni and @jashapiro for input on the plot aesthetics. 

Some other logistical notes: 

- I added to the `sync-data-files.R` some code to grab the alevin and Cell Ranger output files that we need to create these projects. 
- We also need the list of mito genes used, so I created a separate set up script for grabbing any reference files. 
- A lot of the work in this script is reading in the Alevin-fry and Cell Ranger output, making SCE objects, adding cell and gene metrics and then creating the data frames to use for plotting.

jashapiro commented:
I think I like the top version better, with the Cell/Nuclei labels off to the side. 

The only challenge there is that the y axis label is missing, and would be doubled up if you included it. I wonder if just using the facet_wrap() without the cell/nuclei labels is the way to go, but maybe you can color the points by cell/nuclei to add the distinction?

allyhawkins commented:
> The only challenge there is that the y axis label is missing, and would be doubled up if you included it. I wonder if just using the facet_wrap() without the cell/nuclei labels is the way to go, but maybe you can color the points by cell/nuclei to add the distinction?

Do you mean color the facet labels by cell/nuclei?

jashapiro commented:
> > The only challenge there is that the y axis label is missing, and would be doubled up if you included it. I wonder if just using the facet_wrap() without the cell/nuclei labels is the way to go, but maybe you can color the points by cell/nuclei to add the distinction?
> 
> Do you mean color the facet labels by cell/nuclei?

Sorry, yes. Well, sort of. I was thinking you could color the points in the second plot, and the labels in the first? Or maybe the labels in both...

allyhawkins commented:
Okay a few more options:

1. Only label points in correlation plot
![image](https://github.com/AlexsLemonade/scpca-paper-figures/assets/54039191/03d9b590-4b63-4f00-a509-eb2ec35623a2)

![image](https://github.com/AlexsLemonade/scpca-paper-figures/assets/54039191/0c317479-032a-4cf5-b128-7bd9c4cb0f07)
*note that I changed the colors so all 4 are from the same color-blind palette and the cell/nuclei colors would be from the existing suspension palette we have. 

2. Label facets in density plots and color points in correlation plots 
<img width="1293" alt="Screenshot 2024-01-18 at 12 11 42 PM" src="https://github.com/AlexsLemonade/scpca-paper-figures/assets/54039191/dd638789-1137-4790-ad5b-9faa5677da47">

![image](https://github.com/AlexsLemonade/scpca-paper-figures/assets/54039191/acbaefdb-5ad5-4e01-b59c-394ce94af44d)



A few things I figured out: 
- You can't customize facet label colors based on a variable. So to make the version in option 2, I still had to split it, create two separate faceted plots and then combine using `patchwork`. 
- I figured out how to add either a combined x or combined y axis to the patchwork plots, but not both. That's why there's no label on the y-axis for the density plots with colored facet labels. 
- Also because of the label thing, I can't create the correlation plot with differently colored labels unless I split that too and then we have the same x/y axis label problem. 

So after this, I think the cleanest solution would be to keep the labels transparent in all plots, add the `Single-cell` or `Single-nuclei` to the facet label text, and color the points in the correlation plot. If someone knows how to make multiple facet label colors based on a variable in the same plot, then I would be more supportive of coloring labels. 

Thoughts @jashapiro?

jashapiro commented:
> If someone knows how to make multiple facet label colors based on a variable in the same plot, then I would be more supportive of coloring labels.

It looks like it is possible with diving into the internals, but I don't think it is likely to be worth it: https://github.com/tidyverse/ggplot2/issues/2096
I also found the [`ggh4x` package](https://teunbrand.github.io/ggh4x/), which allows coloring facets, but again probably too much work for minimal benefit.

Or, with the patchwork option, it looks like this feature just went in! See https://github.com/thomasp85/patchwork/pull/337 and https://github.com/thomasp85/patchwork/releases/tag/v1.2.0

I'm mostly excited that patchwork is still getting updates. I was worried about it for a bit!

allyhawkins commented:
I had found the `ggh4x` package, but I didn't find a way to color by variable. We could manually set the colors to use though. 

```
backgrounds <- list(element_rect(fill = "#FC8D62"),element_rect(fill = "#FC8D62"), element_rect(fill = "#FC8D62"), 
                    element_rect(fill = "#8DA0CB"),element_rect(fill = "#8DA0CB"),element_rect(fill = "#8DA0CB"))

ggplot(coldata_common, aes(x = sum, color = tool)) + 
      geom_density() + 
      ggh4x::facet_wrap2(~plot_id,
                 scales = "free",
                 strip = ggh4x::strip_themed(background_x = backgrounds)) +
      scale_x_log10(labels = scales::label_number()) +
      labs(x = expression(paste(Log[10], " total UMI per cell")),
           y = "Density", 
           color = "") +
      scale_color_manual(values = method_colors)
```
![image](https://github.com/AlexsLemonade/scpca-paper-figures/assets/54039191/f63f11ba-0466-432d-b9d4-0beae78cc968)

```
gene_exp_plot <- ggplot(rowdata_cor, aes(x = `Alevin-fry`, y = `Cell Ranger`, color = seq_unit)) +
  geom_point(size = 0.5, alpha = 0.1) + 
  ggh4x::facet_wrap2(~plot_id,
                     strip = ggh4x::strip_themed(background_x = backgrounds)) +
  scale_x_log10(labels = scales::label_number()) + 
  scale_y_log10(labels = scales::label_number()) + 
  labs(x = "Cell Ranger mean gene expression", y = "Alevin-fry mean gene expression") + 
  ggpubr::stat_cor(aes(label = after_stat(r.label)), method = "pearson", size = 4) +
  scale_color_manual(values = suspension_colors) +
  theme(legend.position = "none")
```
![image](https://github.com/AlexsLemonade/scpca-paper-figures/assets/54039191/8d4efdd0-4f8d-4a78-8cb9-e939f809753e)

I kind of like these @jashapiro?

jashapiro commented:
> I kind of like these @jashapiro?

They look good to me! Can you change the R to R^2?

allyhawkins commented:
Okay, these should now be ready for a formal review. I'm still saving each plot separately, but I imagine we will combine them with the time/memory plot in illustrator.

jashapiro commented on scripts/FigS1B-D_method-metrics-comparison.R:
```diff
@@ -0,0 +1,223 @@
+umi_plot <- ggplot(coldata_common, aes(x = sum, color = tool)) + 
+  geom_density() + 
+  ggh4x::facet_wrap2(~plot_id,
+                     scales = "free",
```
I think we don't want "free" scales for these plots? I think the scales are close enough to each other to plot everything on the same sets of axes, unless it looks terrible.

jashapiro reviewed (commented):
I mostly had a few little style suggestions, but the main thing I want to see is whether we can use shared axes (not free scales) for the density plots. I think I would prefer that, unless the plots become _really_ unbalanced.

allyhawkins commented:
Thanks for the suggestions @jashapiro! I incorporated your suggestion and removed the `scales = "free"`. I think that was leftover from when I was trying to play around with faceting by two variables. This should be ready for another look.

jashapiro reviewed (commented):
I think this all looks pretty good. My only remaining significant comment is that the font sizes seem small. We could probably fix that by making the output plots themselves smaller, or adjusting the theme. 

I also just wondered if the number of genes detected might be better as a linear plot, but that is just a thought. I could be very wrong, and won't know without seeing it... or if you already looked and it looked terrible.

allyhawkins commented on scripts/FigS1B-D_method-metrics-comparison.R:
```diff
@@ -0,0 +1,218 @@
+  geom_density() + 
+  ggh4x::facet_wrap2(vars(plot_id),
+                     strip = ggh4x::strip_themed(background_x = backgrounds)) +
+  scale_x_log10(labels = scales::label_number()) +
```
For reference this is without the log scale, so I think we do want it. My guess is that when we created the original FAQ figures we chose the log scale for a reason. 

![FigS1C-genes-detected-benchmarking](https://github.com/AlexsLemonade/scpca-paper-figures/assets/54039191/604c8ad7-6e72-4673-9a84-cee273cf92e2)

allyhawkins commented:
I increase the font size all around to be 12, but did not adjust the scale, see https://github.com/AlexsLemonade/scpca-paper-figures/pull/29#discussion_r1459247115.
